src/main/java/com/exasol/releaserobot/repository/Repository.java

```diff
@@ -0,0 +1,24 @@
+package com.exasol.releaserobot.repository;
+
+import java.util.Optional;
+
+/**
+ * Represents a repository where a Git-based project is stored. A repository can be local or remote (GitHub, for
+ * example).
+ */
+public interface Repository {
```

Comment:
this should be a concrete class part of the domain model rather than an interface. Todo in another PR